lib/Connection.php

```diff
@@ -482,110 +472,125 @@ private function validateConfiguration() {
 		return $this->doCriticalValidation();
 	}
 
+	/**
+	 * This function checks the ldap error no and also the LDAP_OPT_DIAGNOSTIC_MESSAGE
+	 * to log as much information as possible. Login attempts with wrong credentials
+	 * will only be logged in debug mode.
+	 *
+	 * @param string $msg The prefix for the log message
+	 * @param array $context
+	 */
+	private function checkError($msg = 'LDAP Error check', $context = ['app' => __METHOD__]) {
+		$errno = $this->getLDAP()->errno($this->ldapConnectionRes);
+		if ($errno === self::LDAP_SUCCESS) {
+			\OC::$server->getLogger()->debug("$msg: ($errno), ".\var_export($this->ldapConnectionRes, true), $context);
+			return;
+		}
+		$errmsg = $this->getLDAP()->error($this->ldapConnectionRes);
+		if (\is_resource($this->ldapConnectionRes)) {
+			$this->getLDAP()->getOption($this->ldapConnectionRes, LDAP_OPT_DIAGNOSTIC_MESSAGE, $errdiag);
```

Comment:
won't work. `$errdiag` should be passed by reference, but `func_get_args` is limited to get the values only https://github.com/owncloud/user_ldap/blob/master/lib/LDAP.php#L282